core/invoker/src/main/scala/whisk/core/containerpool/ContainerProxy.scala

```diff
@@ -380,18 +382,25 @@ class ContainerProxy(
     val activationWithLogs: Future[Either[ActivationLogReadingError, WhiskActivation]] = activation
       .flatMap { activation =>
         val start = tid.started(this, LoggingMarkers.INVOKER_COLLECT_LOGS)
-        collectLogs(tid, job.msg.user, activation, container, job.action)
-          .andThen {
-            case Success(_) => tid.finished(this, start)
-            case Failure(t) => tid.failed(this, start, s"reading logs failed: $t")
-          }
-          .map(logs => Right(activation.withLogs(logs)))
-          .recover {
-            case LogCollectingException(logs) =>
-              Left(ActivationLogReadingError(activation.withLogs(logs)))
-            case _ =>
-              Left(ActivationLogReadingError(activation.withLogs(ActivationLogs(Vector(Messages.logFailure)))))
-          }
+        Try(
+          collectLogs(tid, job.msg.user, activation, container, job.action)
```

Comment:
Where is the exception thrown? Can we make `collectLogs` not throw an exception but rather return a `Future.failed` in that case?

If not: We should only wrap `collectLogs` into a `Try` or just use a `try/catch` here to not box unnecessary.

Reply:
Sorry I'm a total novice with Akka so not sure how to proceed, the error i think is being thrown when Akka tries to "FileTailSource" https://github.com/apache/incubator-openwhisk/blob/85bb3fe38642076646954a8ed1243b76d7f0fee6/core/invoker/src/main/scala/whisk/core/containerpool/docker/DockerClientWithFileAccess.scala#L150
This file isn't readable as the user may not have permissions so it blows up. Still trying to figure out the right way to recover from the error.

Reply:
> Where is the exception thrown? Can we make collectLogs not throw an exception but rather return a Future.failed in that case?

@markusthoemmes  Problem here is that `collectLogs` is provided from outside. So we would need to rely on it returning a Future.failed. More details are provided [here][1]

[1]: https://github.com/apache/incubator-openwhisk/issues/3195#issuecomment-358734535

Reply:
@chetanmeh got it, thanks for the clarification.